### PR TITLE
Update version command to check for new release tag

### PR DESCRIPTION
**Pull request by chndler**
## What was changed

Added logic to version command to check for newer release tag from Github

## Checklist

1. Closes 

2. How was this tested:

3. Any docs updates needed?

### CODE REVIEW

File: app/version.go [hunk 1]
```diff
@@ -109,3 +125,51 @@ func NewBuildInfo() BuildInfo {
 
 	return info
 }
+
+func checkForUpdate(buildInfo *BuildInfo) error {
+	latestInfo, err := fetchLatestVersion()
+	if err != nil {
+		return fmt.Errorf("failed to fetch latest version: %w", err)
```
Comment (chndler) [inline app/version.go:132]: @feihuang after a little more thought I decided to return an error if the client is unable to get the latest version, however the local build info is still printed.

**Comment by feihuang** [comment]
@chndler could you screenshot what the new command output looks like (both when the user is online and offline)?

**Comment by chndler** [comment]
> @chndler could you screenshot what the new command output looks like (both when the user is online and offline)?

@feihuang 
Output when online:
```
$ ./tcld v
{
    "Date": "2024-04-02T15:02:38-07:00",
    "Commit": "4a1386933c59",
    "Version": "v0.20.0"
}
You are running the latest version
```
Output when offline:
```
$ ./tcld v
{
    "Date": "2024-04-02T15:02:38-07:00",
    "Commit": "4a1386933c59",
    "Version": "v0.20.0"
}
failed to fetch latest version: Get "https://api.github.com/repos/temporalio/tcld/releases/latest": dial tcp: lookup api.github.com: no such host
```

**Comment by feihuang** [comment]
> > @chndler could you screenshot what the new command output looks like (both when the user is online and offline)?
> 
> @feihuang Output when online:
> 
> ```
> $ ./tcld v
> {
>     "Date": "2024-04-02T15:02:38-07:00",
>     "Commit": "4a1386933c59",
>     "Version": "v0.20.0"
> }
> You are running the latest version
> ```
> 
> Output when offline:
> 
> ```
> $ ./tcld v
> {
>     "Date": "2024-04-02T15:02:38-07:00",
>     "Commit": "4a1386933c59",
>     "Version": "v0.20.0"
> }
> failed to fetch latest version: Get "https://api.github.com/repos/temporalio/tcld/releases/latest": dial tcp: lookup api.github.com: no such host
> ```
> 
> Output when running tcld built with `go build -o tcld ./cmd/tcld`:
> 
> ```
> $ ./tcld v
> {
>     "Date": "2024-04-02T22:10:27Z",
>     "Commit": "7da7252c0e98a539dc5a4b8cf459109631385eed",
>     "Version": "v0.1.3+no-version-available"
> }
> no-version-available detected, unable to determine if current build is latest version
> ```

When the user is offline, should we just hide the error? It's more a product question - if we have PM for this feature, we should confirm with them. Overall lgtm.

File: app/version.go [hunk 2]
```diff
@@ -109,3 +125,51 @@ func NewBuildInfo() BuildInfo {
...
+}
+
+func fetchLatestVersion() (Release, error) {
+	url := "https://api.github.com/repos/temporalio/tcld/releases/latest"
+	resp, err := http.Get(url)
+	if err != nil {
+		return Release{}, err
+	}
+
+	defer func(Body io.ReadCloser) {
```
Comment (feihuang) [inline app/version.go:153]: nit: Body -> body (golang uses snakeCase naming convention)

**Review by feihuang — APPROVED** [review]

**Comment by chndler** [comment]
> > > @chndler could you screenshot what the new command output looks like (both when the user is online and offline)?
> > 
> > 
> > @feihuang Output when online:
> > ```
> > $ ./tcld v
> > {
> >     "Date": "2024-04-02T15:02:38-07:00",
> >     "Commit": "4a1386933c59",
> >     "Version": "v0.20.0"
> > }
> > You are running the latest version
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Output when offline:
> > ```
> > $ ./tcld v
> > {
> >     "Date": "2024-04-02T15:02:38-07:00",
> >     "Commit": "4a1386933c59",
> >     "Version": "v0.20.0"
> > }
> > failed to fetch latest version: Get "https://api.github.com/repos/temporalio/tcld/releases/latest": dial tcp: lookup api.github.com: no such host
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Output when running tcld built with `go build -o tcld ./cmd/tcld`:
> > ```
> > $ ./tcld v
> > {
> >     "Date": "2024-04-02T22:10:27Z",
> >     "Commit": "7da7252c0e98a539dc5a4b8cf459109631385eed",
> >     "Version": "v0.1.3+no-version-available"
> > }
> > no-version-available detected, unable to determine if current build is latest version
> > ```
> 
> When the user is offline, should we just hide the error? It's more a product question - if we have PM for this feature, we should confirm with them. Overall lgtm.

Reached out to PM that raised this issue and included you.

File: app/version.go (hunk 2 above)
Comment (chndler) [reply app/version.go]: Went ahead and simplified the defer logic.

**Comment by chndler** [comment]
Was informed by PM that : 
> If tcld provides an error, then my guidance is to align to that error message. If tcld doesn’t provide an error, then v should not display an error

So will merge diff with main.

**Pull request merged**